atc/db/build.go

```diff
@@ -1617,7 +1650,7 @@ func updateTransitionBuildForJob(tx Tx, jobID int, buildID int, buildStatus Buil
 		return nil
 	}
 
-	if latestStatus != buildStatus {
+	if latestStatus != buildStatus && (rerunID == 0 || rerunID == latestID) {
```

Comment:
This line started my investigation. I tried this change to discover the failing test:
```suggestion
	if latestStatus != buildStatus && (rerunID == latestID) {
```